website/content/en/docs/building-operators/golang/advanced-topics.md

```diff
@@ -113,10 +113,193 @@ func init() {
 * After adding new import paths to your operator project, run `go mod vendor` if a `vendor/` directory is present in the root of your project directory to fulfill these dependencies.
 * Your 3rd party resource needs to be added before add the controller in `"Setup all Controllers"`.
 
-### Metrics
+### Monitoring and Observability
+This section aims to explain how to expose the first operator's custom metrics and alerts. It focuses on the technical aspects of adding metrics and alerts, and demonstrates it over the sample [memcached-operator]. For more information regarding operators' monitoring, visit [observability-best-practices].  
 
-To learn about how metrics work in the Operator SDK read the [metrics section][metrics_doc] of the Kubebuilder documentation.
+#### Prerequisites
+- Deploy [kube-prometheus] on your cluster. This will set a Prometheus instance on the cluster, which will enable the Prometheus UI, in which we will inspect the metrics, [alerts] and [recording rules].
+- Make sure Prometheus has access to the operator's namespace, by setting the corresponding RBAC rules. You may find [prometheus_role.yaml] and [prometheus_role_binding.yaml] examples helpful. 
+- Create a new `monitoring` directory, which will contain the metrics and alerts implementation. It is recommended to create a dedicated `monitoring` directory, in order to keep the monitoring implementation separated from the core operator's logic. This will allow easier maintenance of both the core operator code and the monitoring code. 
```

Comment:
I think we need to refactor this a bit to be clearer..
Perhaps:

#### Prerequisites

The following are required in order to inspect the metric, [alerts] and [recording rules], in Prometheus, during the development:
- Deploy [kube-prometheus] on your cluster.
- Enable Prometheus access to the required namespace, by setting the RBAC rules.  
  Example:  [prometheus_role.yaml] and [prometheus_role_binding.yaml].
- Create a  `/monitoring` sub-folder in your operator repository.  
**Note:** It is recommended to create a dedicated `/monitoring` directory, in order to keep the monitoring code and the core operator code separated.

Reply:
updated the two prerequisites, based on @varshaprasad96 comments.
Regarding the third one - please see https://github.com/operator-framework/operator-sdk/pull/6341#discussion_r1129835586